Branch-SDK/src/io/branch/referral/Branch.java

```diff
@@ -3142,18 +2045,19 @@ private void setActivityLifeCycleObserver(Application application) {
 
         @Override
         public void onActivityCreated(Activity activity, Bundle bundle) {
+            intentState_ = INTENT_STATE.PENDING;
             if (BranchViewHandler.getInstance().isInstallOrOpenBranchViewPending(activity.getApplicationContext())) {
                 BranchViewHandler.getInstance().showPendingBranchView(activity);
             }
         }
 
         @Override
         public void onActivityStarted(Activity activity) {
+            intentState_ = INTENT_STATE.PENDING;
```

Comment:
Instead of adding this additional latency, can you only do this in Marshmallow or greater? Then any other instance you can just set `INTENT_STATE.READY` in Started.